FastPriorityQueue.js

```diff
@@ -169,7 +204,7 @@ FastPriorityQueue.prototype.peek = function() {
   return this.array[0];
 };
 
-// remove the element on top of the heap (a smallest element)
+// remove the element on top of the heap (the smallest element)
```

Comment:
It is *a* smallest element, because you have no reason to believe that there is a unique smallest element.